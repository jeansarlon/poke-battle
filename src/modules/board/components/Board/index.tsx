import React from 'react';
import { Hand } from '../../../hand/components';

const Board = ({ styleclass, title }: Record<string, string>) => (
  <div className={styleclass}>
    {title}
    <Hand />
  </div>
);

export default Board;
