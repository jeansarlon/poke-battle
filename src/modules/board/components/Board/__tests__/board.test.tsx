import React from 'react';
import { render } from '@testing-library/react';
import Board from '..';

describe('Board', () => {
  test('should renders component', () => {
    const boardClass = 'board';
    const title = 'Board';
    const { container, getByText } = render(<Board styleclass={boardClass} title={title} />);

    expect(getByText(title)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('should has StyleClass', () => {
    const styleClass = 'board';
    const title = 'Board';
    const { getByText } = render(<Board styleclass={styleClass} title={title} />);
    expect(getByText(title)).toHaveClass(styleClass);
  });
});
