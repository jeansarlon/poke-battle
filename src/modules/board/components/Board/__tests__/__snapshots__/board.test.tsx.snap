// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should renders component 1`] = `
<div>
  <div
    class="board"
  >
    Board
    <div
      class="hand"
    >
      <div
        class="cards"
      >
        <span
          class="no-cards"
        >
          MÃ£o vazia
        </span>
      </div>
    </div>
  </div>
</div>
`;
